src/fluree/db/storage/core.cljc

```diff
@@ -337,8 +337,8 @@
 
 (defn resolve-empty-leaf
   [{:keys [comparator] :as node}]
-  (let [pc         (async/promise-chan)
+  (let [ch         (async/chan)
```

Comment:
Out of curiosity, why the move away from promise-chan?

Reply:
@zonotope did this, but I'd assume is because there is only a single take ever on the channel.